lapce-ui/src/editor/tab_header_content.rs

```diff
@@ -351,9 +425,31 @@ impl Widget<LapceTabData> for LapceEditorTabHeaderContent {
 
         self.rects.clear();
         let mut x = 0.0;
+        let mut name_count: HashMap<String, i32> = HashMap::new();
```

Comment:
I think building this hashmap could have a better place in `update`. In general, you want to do it when the data changes, not when the UI changes.